packages/react-dom/src/events/__tests__/DOMModernPluginEventSystem-test.internal.js

```diff
@@ -119,4 +617,375 @@ describe('DOMModernPluginEventSystem', () => {
+    const onMouseLeave = jest.fn(e => log.push(e.currentTarget));
+
+    function Child() {
+      return (
+        <div
+          ref={divRef}
+          onMouseEnter={onMouseEnter}
+          onMouseLeave={onMouseLeave}
+        />
+      );
+    }
+
+    function Parent() {
+      const [portal, setPortal] = React.useState(null);
+
+      React.useLayoutEffect(() => {
+        setPortal(ReactDOM.createPortal(<Child />, portalRef.current));
+      }, []);
+
+      return (
+        <button
+          ref={buttonRef}
+          onMouseEnter={onMouseEnter}
+          onMouseLeave={onMouseLeave}>
+          <div ref={portalRef}>{portal}</div>
+        </button>
+      );
+    }
+
+    ReactDOM.render(<Parent />, container);
+
+    let buttonElement = buttonRef.current;
+    buttonElement.dispatchEvent(
+      new MouseEvent('mouseover', {
+        bubbles: true,
+        cancelable: true,
+        relatedTarget: null,
+      }),
+    );
+    expect(onMouseEnter).toHaveBeenCalledTimes(1);
+    expect(onMouseLeave).toHaveBeenCalledTimes(0);
+    expect(log[0]).toEqual(buttonElement);
+
+    let divElement = divRef.current;
+    buttonElement.dispatchEvent(
+      new MouseEvent('mouseout', {
+        bubbles: true,
+        cancelable: true,
+        relatedTarget: divElement,
+      }),
+    );
+    divElement.dispatchEvent(
+      new MouseEvent('mouseover', {
+        bubbles: true,
+        cancelable: true,
+        relatedTarget: buttonElement,
+      }),
+    );
+    expect(onMouseEnter).toHaveBeenCalledTimes(2);
+    expect(onMouseLeave).toHaveBeenCalledTimes(0);
+    expect(log[1]).toEqual(divElement);
+  });
+
+  it('should preserve bubble/capture order between roots and nested portals', () => {
+    const targetRef = React.createRef();
+    let log = [];
+    const onClickRoot = jest.fn(e => log.push('bubble root'));
+    const onClickCaptureRoot = jest.fn(e => log.push('capture root'));
+    const onClickPortal = jest.fn(e => log.push('bubble portal'));
+    const onClickCapturePortal = jest.fn(e => log.push('capture portal'));
+
+    function Portal() {
+      return (
+        <div
+          onClick={onClickPortal}
+          onClickCapture={onClickCapturePortal}
+          ref={targetRef}>
+          Click me!
+        </div>
+      );
+    }
+
+    const portalContainer = document.createElement('div');
+
+    let shouldStopPropagation = false;
+    portalContainer.addEventListener(
+      'click',
+      e => {
+        if (shouldStopPropagation) {
+          e.stopPropagation();
+        }
+      },
+      false,
+    );
+
+    function Root() {
+      let portalTargetRef = React.useRef(null);
+      React.useLayoutEffect(() => {
+        portalTargetRef.current.appendChild(portalContainer);
+      });
+      return (
+        <div onClick={onClickRoot} onClickCapture={onClickCaptureRoot}>
+          <div ref={portalTargetRef} />
+          {ReactDOM.createPortal(<Portal />, portalContainer)}
```

Comment:
Portal are designed so that if I switch this from this to just `<Portal />` then the behavior is preserved. Then it's more obvious why stopPropagation on the sibling doesn't stop it.